h2o-docs/src/product/explainability.rst

```diff
@@ -0,0 +1,205 @@
+
+        # Identify predictors and response
+        y <- "response"
+        x <- setdiff(names(train), y)
+
+        # For binary classification, response should be a factor
+        train[, y] <- as.factor(train[, y])
+        test[, y] <- as.factor(test[, y])
+
+        # Run AutoML for 20 base models
+        aml <- h2o.automl(x = x, y = y, 
+                          training_frame = train,
+                          max_models = 20,
+                          seed = 1)
+
+        # Explain all AutoML models                  
+        exa <- h2o.explain(aml, test)
+
+        # Explain a single model (top AutoML model)                 
+        exm <- h2o.explain(aml@leader, test)         
+
+
+
+   .. code-tab:: python
+
+        import h2o
+        from h2o.automl import H2OAutoML
+
+        h2o.init()
+
+        # Import a sample binary outcome train/test set into H2O
+        train = h2o.import_file("https://s3.amazonaws.com/erin-data/higgs/higgs_train_10k.csv")
+        test = h2o.import_file("https://s3.amazonaws.com/erin-data/higgs/higgs_test_5k.csv")
+
+        # Identify predictors and response
+        x = train.columns
+        y = "response"
+        x.remove(y)
+
+        # For binary classification, response should be a factor
+        train[y] = train[y].asfactor()
+        test[y] = test[y].asfactor()
+        
+        # Run AutoML for 20 base models
+        aml = H2OAutoML(max_models=20, seed=1)
+        aml.train(x=x, y=y, training_frame=train)
+
+        # Explain all models
+        aml.explain(test)
+
+        # Explain a single model
+        aml.leader.explain(test)
+
+
+The code above is the quickest way to get started. 
+
+Explain a single row prediction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``h2o.explain_row()`` function provides model explanations for a single row of test data. 
+
+Using the previous code example, you can evaluate row-level behavior by specifying the `row_index`:
+
+.. tabs::
+   .. code-tab:: r R
+
+        # Explain row 1 with all AutoML models
+        h2o.explain_row(aml, test, row_index = 1)
+
+        # Explain row 1 with a single model
+        h2o.explain_row(aml@leader, row_index = 1)
+
+   .. code-tab:: python
+
+        # Explain row 1 with all AutoML models
+        aml.explain_row(test, row_index=1)
+
+        # Explain row 1 with a single model
+        aml.leader.explain_row(test, row_index=1)
+
+
+Output: Explainations
+---------------------
+
+Explaination Plotting Functions 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TO DO: Let's put examples of each function and the plot, in the order in which they appear in the ``h2o.explain()`` output.  Let's also show how to customize the plots.
+
+.. tabs::
+   .. code-tab:: r R
+
+        # Residual analysis plot for an AutoML object
+        ra_plot <- h2o.residual_analysis(aml, test)
+        ra_plot
+
+   .. code-tab:: python
+
+        # Residual analysis plot for an AutoML object
+        ra_plot = aml.residual_analysis(test)
```

Comment:
Should be `ra_plot = aml.leader.residual_analysis(test)`.

The residual_analysis in python works on subclasses of `H2ORegressionModel` which H2OAutoML isn't AFAIK even if we do regression problem.

I can change the utility functions to be able to be invoked on H2OAutoML if you think it would be better.

Reply:
It's fine to take single model for now, thanks!